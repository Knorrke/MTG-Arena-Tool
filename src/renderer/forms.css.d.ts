/* eslint-disable */
// GENERATED FILE; DO NOT EDIT
declare namespace FormsCssModule {
  export interface IFormsCss {
    "form-authenticate": string;
    "form-button": string;
    "form-container": string;
    "form-error": string;
    "form-icon": string;
    "form-input-container": string;
    "form-label": string;
    "form-options": string;
    formAuthenticate: string;
    formButton: string;
    formContainer: string;
    formError: string;
    formIcon: string;
    formInputContainer: string;
    formLabel: string;
    formOptions: string;
  }
}

declare const FormsCssModule: FormsCssModule.IFormsCss & {
  /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
  locals: FormsCssModule.IFormsCss;
};

export = FormsCssModule;
