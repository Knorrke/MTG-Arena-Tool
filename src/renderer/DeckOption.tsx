/*
  Not used / deprecated?
*/
import React from "react";
import { getDeckName, deckExists } from "../shared/store";
import indexCss from "./index.css";
import sharedCss from "../shared/shared.css";
import { WHITE, BLUE, BLACK, RED, GREEN, COLORLESS } from "../shared/constants";

const manaClasses: string[] = [];
manaClasses[WHITE] = sharedCss.manaW;
manaClasses[BLUE] = sharedCss.manaU;
manaClasses[BLACK] = sharedCss.manaB;
manaClasses[RED] = sharedCss.manaR;
manaClasses[GREEN] = sharedCss.manaG;
manaClasses[COLORLESS] = sharedCss.manaC;

export interface DeckOptionDeck {
  colors?: number[];
  name?: string;
  archived?: boolean;
}

export interface DeckOptionProps {
  deckId: string;
  deck: DeckOptionDeck;
}

export default function DeckOption(props: DeckOptionProps): JSX.Element {
  const { deckId, deck } = props;

  const exists = deckExists(deckId);
  const deckName: string = exists ? getDeckName(deckId) : deck.name || "";
  let maxChars = 10;
  if (exists && deck.colors) {
    maxChars = 16 - 2 * deck.colors.length;
  }

  return (
    <>
      {deckName.length > maxChars ? (
        <abbr title={deckName}>{deckName.slice(0, maxChars)}...</abbr>
      ) : (
        deckName
      )}
      {exists ? (
        <>
          {deck.archived && (
            <small>
              <i>(archived)</i>
            </small>
          )}
          <div className={indexCss.flexItem}>
            {deck.colors &&
              deck.colors.map((color) => (
                <div
                  className={sharedCss.manaS16 + " " + manaClasses[color - 1]}
                  key={color}
                />
              ))}
          </div>
        </>
      ) : (
        <small>
          <i>(deleted)</i>
        </small>
      )}
    </>
  );
}
