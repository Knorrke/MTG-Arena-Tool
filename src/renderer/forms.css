.form-input-container {
  display: flex;
  height: 32px;
  width: 100%;
  padding: 4px 0 12px 0;
}

.form-input-container input {
  background-color: var(--color-mid-40);
  border: 1px solid var(--color-mid-50);
  display: inline-table;
  border-radius: 3px;
  color: var(--color-back);
  height: 22px;
  width: 100%;
  font-size: 15px;
  padding: 6px;
  user-select: none;
  font-family: var(--main-font-name);
}

.form-input-container input:focus {
  border: 1px solid white;
  outline-style: none;
}

.form-label {
  font-family: var(--main-font-name);
  color: var(--color-back);
  margin: auto 0;
}

.form-options {
  width: 18.9em;
  margin: 0 auto auto auto;
  z-index: 5;
  display: flex;
  flex-direction: column;
}

.form-container {
  margin: auto;
  flex-grow: 2;
  display: flex;
  width: -webkit-fill-available;
  height: -webkit-fill-available;
  flex-direction: column;
}

.form-icon {
  background-image: url(../assets/icons/icon-256.png);
  background-color: var(--color-light);
  border-radius: 7em;
  background-size: contain;
  width: 7em;
  height: 7em;
  position: relative;
  margin: auto auto -7em auto;
  top: -7em;
}

.form-authenticate {
  max-width: 18.9em;
  z-index: 5;
  width: -webkit-fill-available;
  margin: auto auto 16px auto;
  background-color: var(--color-light);
  padding: 3.5em 2em 2em 2em;
  border-radius: 3px;
  box-shadow: 0px 2px 7px 4px rgba(0, 0, 0, 0.3);
}

.form-button {
  align-self: center;
  cursor: pointer;
  width: -webkit-fill-available;
  width: -moz-available;
  height: 36px;
  border-radius: 3px;
  color: var(--color-light);
  font-size: 16px;
  text-align: center;
  border: 1px solid var(--color-mid-50);
  background-color: var(--color-back);
  -webkit-transition: all .1s ease-in;
  font-family: var(--main-font-name);
  display: block;
  margin: auto;
}

.form-button:disabled {
  background-color: var(--color-mid-50);
  color: var(--color-light-50);
}

.form-button:hover:enabled {
  background-color: var(--color-mid-66);
  box-shadow: 0px 6px 15px 0px rgba(0, 0, 0, 0.1);
  border: 1px solid var(--color-mid-50);
}

.form-button:active:enabled {
  background-color: var(--color-mid);
  line-height: 34px;
  box-shadow: 0px 3px 8px 0px rgba(0, 0, 0, 0.6);
}

.form-error {
  text-align: center;
  line-height: 32px;
  font-family: var(--main-font-name-bold-it);
  color: var(--color-r);
  width: 19em;
  height: 32px;
  margin: 0px auto;
  -webkit-transition: all .3s ease-in;
}