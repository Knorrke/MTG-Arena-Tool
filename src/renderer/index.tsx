import "../shared/shared.css";
import "./index.css";
import "@github/time-elements";
import RenderApp from "./app/renderApp";

RenderApp();
