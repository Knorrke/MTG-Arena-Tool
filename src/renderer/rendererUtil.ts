/* eslint-disable @typescript-eslint/no-use-before-define, @typescript-eslint/camelcase */
import { app, ipcRenderer as ipc, remote } from "electron";
import path from "path";
import {
  IPC_BACKGROUND,
  IPC_RENDERER,
  CARD_RARITIES,
} from "../shared/constants";
import { WinLossGate } from "../types/event";
import store from "../shared/redux/stores/rendererStore";
import { MissingWildcards, CardCounts } from "./components/decks/types";
import Deck from "../shared/deck";
import db from "../shared/database";

import sharedCss from "../shared/shared.css";

export const actionLogDir = path.join(
  (app || remote.app).getPath("userData"),
  "actionlogs"
);

export function ipcSend(
  method: string,
  arg?: unknown,
  to = IPC_BACKGROUND
): void {
  ipc.send("ipc_switch", method, IPC_RENDERER, arg, to);
}

export function toggleArchived(id: string | number): void {
  ipcSend("toggle_archived", id);
}

export function getTagColor(tag?: string): string {
  return (
    (tag ? store.getState().playerdata.tagsColors[tag] : undefined) ?? "#FAE5D2"
  );
}

export function formatPercent(
  value: number,
  config = { maximumSignificantDigits: 2 }
): string {
  return value.toLocaleString([], {
    style: "percent",
    ...config,
  });
}

export function formatWinrateInterval(lower: number, upper: number): string {
  return `${formatPercent(lower)} to ${formatPercent(upper)} with 95% confidence
(estimated actual winrate bounds, assuming a normal distribution)`;
}

export function formatNumber(value: number, config = {}): string {
  return value.toLocaleString([], {
    style: "decimal",
    ...config,
  });
}

export function getWinrateClass(wr: number, bright = true): string {
  if (wr > 0.65) return bright ? sharedCss.blueBright : sharedCss.blue;
  if (wr > 0.55) return bright ? sharedCss.greenBright : sharedCss.green;
  if (wr < 0.45) return bright ? sharedCss.orangeBright : sharedCss.orange;
  if (wr < 0.35) return bright ? sharedCss.redBright : sharedCss.red;
  return bright ? sharedCss.whiteBright : sharedCss.white;
}

export function getEventWinLossClass(
  wlGate: Partial<WinLossGate>,
  bright = true
): string {
  if (wlGate === undefined)
    return bright ? sharedCss.whiteBright : sharedCss.white;
  if (wlGate.MaxWins === wlGate.CurrentWins)
    return bright ? sharedCss.blueBright : sharedCss.blue;
  if (wlGate.CurrentWins !== undefined && wlGate.CurrentLosses !== undefined) {
    if (wlGate.CurrentWins > wlGate.CurrentLosses)
      return bright ? sharedCss.greenBright : sharedCss.green;
    if (wlGate.CurrentWins * 2 > wlGate.CurrentLosses)
      return bright ? sharedCss.orangeBright : sharedCss.orange;
  }
  return bright ? sharedCss.redBright : sharedCss.red;
}

interface Winrate {
  wins: number;
  losses: number;
  colors?: number[] | undefined;
}

export function compareWinrates(a: Winrate, b: Winrate): -1 | 0 | 1 {
  const _a = a.wins / a.losses;
  const _b = b.wins / b.losses;

  if (_a < _b) return 1;
  if (_a > _b) return -1;

  return compareColorWinrates(a, b);
}

export function compareColorWinrates(winA: Winrate, winB: Winrate): -1 | 0 | 1 {
  const a = winA.colors ?? [];
  const b = winB.colors ?? [];

  if (a.length < b.length) return -1;
  if (a.length > b.length) return 1;

  const sa = a.reduce(function (_a: number, _b: number) {
    return _a + _b;
  }, 0);
  const sb = b.reduce(function (_a: number, _b: number) {
    return _a + _b;
  }, 0);
  if (sa < sb) return -1;
  if (sa > sb) return 1;

  return 0;
}

export function getBoosterCountEstimate(
  neededWildcards: MissingWildcards
): number {
  let boosterCost = 0;
  const boosterEstimates = {
    common: 3.36,
    uncommon: 2.6,
    rare: 5.72,
    mythic: 13.24,
  };

  const playerEconomy = store.getState().playerdata.economy;

  const ownedWildcards = {
    common: playerEconomy.wcCommon,
    uncommon: playerEconomy.wcUncommon,
    rare: playerEconomy.wcRare,
    mythic: playerEconomy.wcMythic,
  };

  CARD_RARITIES.map((rarity) => {
    if (rarity !== "land") {
      const needed = neededWildcards[rarity] || 0;
      const owned = ownedWildcards[rarity] || 0;
      const missing = Math.max(0, needed - owned);
      boosterCost = Math.max(boosterCost, boosterEstimates[rarity] * missing);
    }
  });

  return Math.round(boosterCost);
}

export function getWildcardsMissing(
  deck: Deck,
  grpid: number,
  isSideboard?: boolean
): number {
  let mainQuantity = 0;
  const cards = store.getState().playerdata.cards;
  const mainMatches = deck
    .getMainboard()
    .get()
    .filter((card) => card.id == grpid);
  if (mainMatches.length) {
    mainQuantity = mainMatches[0].quantity;
  }

  let sideboardQuantity = 0;
  const sideboardMatches = deck
    .getSideboard()
    .get()
    .filter((card) => card.id == grpid);
  if (sideboardMatches.length) {
    sideboardQuantity = sideboardMatches[0].quantity;
  }

  let needed = mainQuantity;
  if (isSideboard) {
    needed = sideboardQuantity;
  }
  // cap at 4 copies to handle petitioners, rat colony, etc
  needed = Math.min(4, needed);

  const card = db.card(grpid);
  let arr = [];
  if (!card?.reprints) arr = [grpid];
  else arr.push(grpid);

  let have = 0;
  arr.forEach((id) => {
    const n = cards.cards[id];
    if (n !== undefined) {
      have += n;
    }
  });

  let copiesLeft = have;
  if (isSideboard) {
    copiesLeft = Math.max(0, copiesLeft - mainQuantity);

    const infiniteCards = [67306, 69172]; // petitioners, rat colony, etc
    if (have >= 4 && infiniteCards.indexOf(grpid) >= 0) {
      copiesLeft = 4;
    }
  }

  return Math.max(0, needed - copiesLeft);
}

export function getCardsMissingCount(deck: Deck, grpid: number): number {
  const mainMissing = getWildcardsMissing(deck, grpid, false);
  const sideboardMissing = getWildcardsMissing(deck, grpid, true);
  return mainMissing + sideboardMissing;
}

export function get_deck_missing(deck: Deck): MissingWildcards {
  const missing = { rare: 0, common: 0, uncommon: 0, mythic: 0 };
  const alreadySeenIds = new Set(); // prevents double counting cards across main/sideboard
  const entireDeck = [
    ...deck.getMainboard().get(),
    ...deck.getSideboard().get(),
  ];

  entireDeck.forEach((card) => {
    const grpid = card.id;
    // process each card at most once
    if (alreadySeenIds.has(grpid)) {
      return;
    }
    const rarity = db.card(grpid)?.rarity;
    if (rarity && rarity !== "land") {
      missing[rarity] += getCardsMissingCount(deck, grpid);
      alreadySeenIds.add(grpid); // remember this card
    }
  });

  return missing;
}

export function getMissingCardCounts(deck: Deck): CardCounts {
  const missingCards: CardCounts = {};
  const allCardIds = new Set(
    [...deck.getMainboard().get(), ...deck.getSideboard().get()].map(
      (card) => card.id
    )
  );
  allCardIds.forEach((grpid) => {
    const missing = getCardsMissingCount(deck, grpid);
    if (missing > 0) {
      missingCards[grpid] = missing;
    }
  });
  return missingCards;
}
