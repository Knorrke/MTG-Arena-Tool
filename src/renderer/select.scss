
.select_container {
  width: 240px;
  height: 32px;
  display: flex;
  flex-direction: column;
  position: relative;
  margin: auto 8px;
}

.select_title {
  white-space: nowrap;
  background-color: var(--color-back);
  font-family: var(--main-font-name-bold);
  opacity: 0.66;
  height: 33px;
  font-size: 100%;
  padding: 0 12px;
  justify-content: center;
  line-height: 33px;
}

.select_button {
  display: flex;
  color: var(--color-light);
  overflow: hidden;
  text-indent: 10px;
  cursor: pointer;
  width: inherit;
  height: 32px;
  border-radius: 4px;
  background-color: var(--color-mid-20);
  border: 1px solid var(--color-light-50);
  -webkit-transition: all .1s cubic-bezier(0.2, 0.5, 0.35, 1);
  justify-content: space-between;
  align-items: center;
  white-space: nowrap;
  margin: 0;
  padding: 0;

  .light {
    color: var(--color-back);
    background-color: var(--color-mid-40);
    border: 1px solid var(--color-mid-50);
  }
}

.select_button:after {
  content: "";
  width: 0px;
  height: 0;
  border: 7px solid transparent;
  border-color: var(--color-light) transparent transparent transparent;
  position: absolute;
  top: 13px;
  right: 8px;
}

.select_button:hover:enabled {
  background-color: var(--color-mid-50);
  border: 1px solid var(--color-light);
}

.select_button.active {
  border-radius: 4px 4px 0px 0px;
  border-bottom: 0px;
  background-color: var(--color-mid-50);
}

.select_button.active:after {
  top: 5px;
  border-color: transparent transparent var(--color-light) transparent
}

.select_options_container {
  color: var(--color-light);
  cursor: pointer;
  position: absolute;
  max-height: 384px;
  overflow-y: auto;
  overflow-x: hidden;
  top: calc(100% + 0px);
  right: 0px;
  left: 0px;
  z-index: 999;
  border-radius: 0 0 4px 4px;
  background-color: hsla(29, 22%, 30%, 1);
  border: 1px solid var(--color-mid);
  border-top: 0px;
  box-shadow: rgba(0, 0, 0, 0.3) 0px 6px 8px 4px;

  .light {
    background-color: var(--color-light);
    box-shadow: rgba(0, 0, 0, 0.1) 0px 6px 8px 4px;
  }
}

.select_option {
  cursor: pointer;
  display: flex;
  justify-content: space-between;
  align-items: center;
  white-space: nowrap;
  color: var(--color-light);
  margin: 0;
  height: 33px;
  padding: 0 12px;
  /*border-top: 1px solid var(--color-mid-50);*/
  -webkit-transition: all .1s cubic-bezier(0.2, 0.5, 0.35, 1);
  width: 100%;

  .light {
    color: var(--color-light);
    background-color: var(--color-mid-66);
  }
}

.select_option:hover {
  .light {
    background-color: var(--color-mid-75);
  }
}

.select_option:disabled {
  background: var(--color-back);

  .light {
    background: var(--color-mid-75);
  }
}

.select_option:hover:enabled, .ASsuggestionHighlighted {
  background: var(--color-back-40);

  .light {
    background-color: var(--color-mid-75);
  }
}

.select-hidden {
  display: none;
  visibility: hidden;
  padding-right: 10px;
}

.select {
  cursor: pointer;
  display: inline-block;
  position: relative;
  font-size: 14px;
  color: var(--color-light);
  width: 240px;
  height: 32px;
  margin: auto 0;
}

.select-styled {
  position: absolute;
  border-radius: 6px;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-color: var(--color-mid-20);
  border: 1px solid var(--color-light-50);
  padding: 8px 15px;
  -webkit-transition: all .1s cubic-bezier(0.2, 0.5, 0.35, 1);
}

.select-styled:after {
  content:"";
  width: 0;
  height: 0;
  border: 7px solid transparent;
  border-color: var(--color-light) transparent transparent transparent;
  position: absolute;
  top: 12px;
  right: 10px;
}

.select-styled:hover {
  background-color: var(--color-mid-50);
  border: 1px solid var(--color-light);
}

.select-styled:active, .select-styled.active {
  border-radius: 6px 6px 0px 0px;
  border-bottom: 0px;
  background-color: var(--color-mid-50);
}

.select-styled:active:after, .select-styled.active:after {
  top: 5px;
  border-color: transparent transparent var(--color-light) transparent
}

.select-styled.active + .select-options {
  display: block;
}

.select-options {
  display: none;
  position: absolute;
  max-height: 384px;
  overflow-y: auto;
  overflow-x: hidden;
  top: 100%;
  right: 0;
  left: 0;
  z-index: 999;
  margin: 0;
  padding: 0;
  list-style: none;
  background-color: hsla(29, 22%, 30%, 1);
  border: 1px solid var(--color-light-50);
  border-top: 0px;
  box-shadow: rgba(0, 0, 0, 0.25) 0px 10px 16px 0px;

  li {
    margin: 0;
    padding: 7px 0;
    text-indent: 15px;
    border-top: 1px solid var(--color-mid-50);
    -webkit-transition: all .25s cubic-bezier(0.2, 0.5, 0.35, 1);
  }
  
  li:hover {
    background: var(--color-light-50);
  }
  
  li[rel="hide"] {
    display: none;
  }
}

.settings-select {
  display: flex;
  align-items: center;
}

.AScontainer {
}

.AScontainerOpen {
}

.ASinput {
  border: none;
  height: 17px;
  background-color: inherit;
}

.ASsuggestionsList {
  color: var(--color-light);
  cursor: pointer;
  max-height: 384px;
  overflow-y: auto;
  top: 21px;
  right: 0px;
  left: 0px;
  z-index: 999;
  margin: -2px;
  padding: 0 4px;
  background-color: var(--color-mid-40);
  border: 1px solid var(--color-light-50);
  border-top: 0px;
  box-shadow: rgba(0, 0, 0, 0.25) 0px 10px 16px 0px;
  width: 180px;
}

.ASsuggestion {
  list-style-type: none;
}

.ASsuggestionFirst {
}

.ASsectionContainer {
}

.ASsectionContainerFirst {
}

.ASsectionTitle {
}
