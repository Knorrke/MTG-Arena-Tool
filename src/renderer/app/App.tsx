import { remote } from "electron";
import anime from "animejs";
import React, { useEffect, useRef } from "react";
import { useDispatch, useSelector } from "react-redux";
import {
  LOGIN_WAITING,
  LOGIN_OK,
  IPC_NONE,
  EASING_DEFAULT,
} from "../../shared/constants";
import ErrorBoundary from "./ErrorBoundary";
import { TopNav } from "../components/main/topNav";
import { forceOpenAbout, getOpenNav, getOpenSub } from "../tabControl";
import BackgroundImage from "../components/main/BackgroundImage";
import TopBar from "../components/main/TopBar";
import LoadingBar from "../components/main/LoadingBar";
import Auth from "../components/main/Auth";
import ipcListeners from "./ipcListeners";
import Popup from "../components/main/Popup";
import CardHover from "../components/main/CardHover";
import OutputLogInput from "../components/popups/OutputLogInput";
import { ipcSend } from "../rendererUtil";
import Share from "../components/popups/Share";
import store, { AppState } from "../../shared/redux/stores/rendererStore";
import { reduxAction } from "../../shared/redux/sharedRedux";
import initializeRendererReduxIPC from "../../shared/redux/initializeRendererReduxIPC";

import css from "./app.css";
import AuthSettings from "../components/auth-settings";

initializeRendererReduxIPC(store);

export function App(): JSX.Element {
  const loginState = useSelector((state: AppState) => state.login.loginState);
  const openAuthSettings = useSelector(
    (state: AppState) => state.renderer.authSettings
  );
  const topArtist = useSelector((state: AppState) => state.renderer.topArtist);
  const offline = useSelector((state: AppState) => state.renderer.offline);
  const loading = useSelector((state: AppState) => state.renderer.loading);
  const navIndex = useSelector((state: AppState) => state.renderer.navIndex);
  const topNav = useSelector((state: AppState) => state.renderer.topNav);
  const subNavType = useSelector(
    (state: AppState) => state.renderer.subNav.type
  );
  const subNavId = useSelector((state: AppState) => state.renderer.subNav.id);
  const subNavData = useSelector(
    (state: AppState) => state.renderer.subNav.data
  );
  const authForm = useSelector((state: AppState) => state.login.loginForm);
  const noLog = useSelector((state: AppState) => state.renderer.noLog);
  const share = useSelector((state: AppState) => state.renderer.shareDialog);
  const movingUxRef = useRef<HTMLDivElement | null>(null);
  /*
    Set up IPC listeners.
    This should only happen once when the app starts, so no
    action should recreate the App component.
    IPC Listeners should be inside a React component below the
    context provider hierarchy, so they can dispatch actions.
  */
  const dispatch = useDispatch();
  React.useEffect(() => {
    ipcListeners(dispatch);
  }, [dispatch]);

  const closeNoLog = React.useCallback(
    (log: string) => {
      setTimeout(() => {
        reduxAction(dispatch, { type: "SET_NO_LOG", arg: false }, IPC_NONE);
        ipcSend("set_log", log);
      }, 350);
    },
    [dispatch]
  );

  const closeShare = React.useCallback(() => {
    setTimeout(() => {
      reduxAction(
        dispatch,
        { type: "SET_SHARE_DIALOG_OPEN", arg: false },
        IPC_NONE
      );
    }, 350);
  }, [dispatch]);

  const closeSettings = React.useCallback(() => {
    setTimeout(() => {
      reduxAction(
        dispatch,
        { type: "SET_AUTH_SETTINGS", arg: false },
        IPC_NONE
      );
    }, 350);
  }, [dispatch]);

  useEffect(() => {
    setTimeout(() => {
      anime({
        targets: movingUxRef.current, //css.movingUx,
        left: navIndex * -100 + "%",
        easing: EASING_DEFAULT,
        duration: 350,
      });
    }, 10);
  }, [navIndex, movingUxRef]);

  return (
    <>
      <BackgroundImage />
      <div className={css.outer_wrapper}>
        <TopBar artist={topArtist} offline={offline} />
        <Popup />
        {noLog ? <OutputLogInput closeCallback={closeNoLog} /> : <></>}
        {share.open ? <Share closeCallback={closeShare} /> : <></>}
        <CardHover />
        {loginState == LOGIN_OK ? <TopNav /> : <></>}
        {loading || loginState == LOGIN_WAITING ? (
          <LoadingBar style={loginState == LOGIN_OK ? { top: "99px" } : {}} />
        ) : (
          <></>
        )}
        <ErrorBoundary>
          {loginState == LOGIN_OK ? (
            <div className={css.wrapper}>
              <div className={css.overflowUxMain}>
                <div className={css.movingUx} ref={movingUxRef}>
                  {getOpenNav(topNav, offline)}
                  <div className={css.uxItem}>
                    {getOpenSub(subNavType, subNavId, subNavData)}
                  </div>
                  <div className={css.uxItem}></div>
                </div>
              </div>
            </div>
          ) : (
            <>
              <Auth authForm={authForm} />
              {openAuthSettings && (
                <AuthSettings closeCallback={closeSettings} />
              )}
            </>
          )}
        </ErrorBoundary>
      </div>
      <div className={css.version_number} onClick={forceOpenAbout}>
        v{remote.app.getVersion()}
      </div>
    </>
  );
}

export default App;
