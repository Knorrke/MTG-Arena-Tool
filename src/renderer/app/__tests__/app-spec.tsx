/* eslint-env jest */
import renderer from "react-test-renderer";

import React from "react";
import { App } from "../App";
import { Provider } from "react-redux";
import store from "../../../shared/redux/stores/rendererStore";

describe("App component", () => {
  it("renders properly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <App />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
