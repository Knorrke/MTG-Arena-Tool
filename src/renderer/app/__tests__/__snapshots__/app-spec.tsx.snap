// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders properly 1`] = `
Array [
  <div
    className="main_wrapper main_bg_image"
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.3)",
        "backgroundImage": "../assets/images/Bedevil-Art.jpg",
      }
    }
  >
    <div
      className="wrapper_after"
    />
  </div>,
  <div
    className="outer_wrapper"
  >
    <div
      className="top"
    >
      <div
        className="flex_item"
      >
        <div
          className="top_logo"
        />
        <div
          className="top_artist"
        >
          Bedevil by Seb McKinnon
        </div>
      </div>
      <div
        className="flex_item"
      >
        <div
          className="button minimize"
          onClick={[Function]}
        />
        <div
          className="button settings"
          onClick={[Function]}
        />
        <div
          className="button close"
          onClick={[Function]}
        />
      </div>
    </div>
    <div
      className="popup"
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      
    </div>
    <div
      className="card-hover-dfc"
      style={
        Object {
          "backgroundImage": "url()",
          "height": "348.1481481495923px",
          "opacity": 0,
          "right": "298px",
          "top": "calc(100% - 380.1481481495923px)",
          "width": "250px",
        }
      }
    />
    <div
      className="card-hover-main"
      style={
        Object {
          "backgroundImage": "url()",
          "height": "348.1481481495923px",
          "opacity": 0,
          "top": "calc(100% - 380.1481481495923px)",
          "width": "250px",
        }
      }
    />
    <div
      className="form-container"
    >
      <div
        className="form-authenticate"
      >
        <div
          className="form-icon"
        />
        <div
          id="loginform"
        >
          <label
            className="form-label"
          >
            Email
          </label>
          <div
            className="form-input-container"
          >
            <input
              autoComplete="off"
              id="signin_email"
              onChange={[Function]}
              type="email"
              value=""
            />
          </div>
          <label
            className="form-label"
          >
            Password
          </label>
          <div
            className="form-input-container"
          >
            <input
              autoComplete="off"
              id="signin_pass"
              onChange={[Function]}
              type="password"
              value=""
            />
          </div>
          <div
            style={
              Object {
                "color": "var(--color-mid-75)",
                "cursor": "pointer",
                "marginBottom": "16px",
              }
            }
          >
            <a
              className="forgot_link"
              onClick={[Function]}
            >
              Forgot your password?
            </a>
          </div>
          <button
            className="form-button"
            disabled={false}
            id="submit"
            onClick={[Function]}
            type="submit"
          >
            Login
          </button>
          <div
            className="form-error"
          >
            
          </div>
        </div>
      </div>
      <div
        className="form-options"
      >
        <label
          className="check_container hover_label"
          onClick={[Function]}
          style={
            Object {
              "margin": "auto auto 12px auto",
              "width": "max-content",
            }
          }
        >
          Remember me?
          <input
            checked={false}
            disabled={true}
            type="checkbox"
          />
          <span
            className="checkmark"
          />
        </label>
        <div
          className="message_small"
        >
          Dont have an account?
           
          <a
            className="signup_link"
            onClick={[Function]}
          >
            Sign up!
          </a>
        </div>
        <div
          className="message_small"
        >
          You can also
           
          <a
            className="offline_link"
            onClick={[Function]}
          >
            continue offline
          </a>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="version_number"
    onClick={[Function]}
  >
    v
    1.0.0
  </div>,
]
`;
